main.go

```diff
@@ -110,14 +155,153 @@ func runMain() error {
 		return stacktrace.Propagate(err, "An error occurred parsing revision '%v'", remoteMasterBranchName)
 	}
 
-	fmt.Println(localMasterHash.String())
-	fmt.Println(remoteMasterHash.String())
+	isLocalMasterInSyncWithRemoteMaster := localMasterHash.String() == remoteMasterHash.String()
+	if !isLocalMasterInSyncWithRemoteMaster {
+		fmt.Println("The local master branch is not in sync with the remote master branch. Must be in sync to conduct release process.")
+		return nil
+	}
+
+	fmt.Println("Finished prererelease checks.")
+	
+	fmt.Println("Guessing next release version...")
+	latestReleaseVersion := getLatestReleaseVersion(repository, NO_PREVIOUS_VERSION)
+
+	// Conduct changelog file validation
+	changelogFilepath := path.Join(currentWorkingDirectory, relChangelogFilepath)
+	tbdHeaderCount := grepFile(changelogFilepath, TBD_VERSION_HEADER_REGEX)
+	if tbdHeaderCount != EXPECTED_NUM_TBD_HEADER_LINES {
+		fmt.Printf("There should be %d TBD header lines in the changelog. Instead there are %d.\n", EXPECTED_NUM_TBD_HEADER_LINES, tbdHeaderCount)
+		return nil
+	}
+	versionHeaderCount := grepFile(changelogFilepath, VERSION_HEADER_REGEX)
+	if versionHeaderCount == 0 {
+		fmt.Println("No previous changelog versions were detected in this changelog. Are you sure that the changelog is in sync with the release tags on this branch?")
+		return nil
+	}
+
+	existsBreakingChanges := detectBreakingChanges(changelogFilepath)
+	var nextReleaseVersion semver.Version
+	if existsBreakingChanges {
+		nextReleaseVersion = latestReleaseVersion.IncMinor()
+	} else {
+		nextReleaseVersion = latestReleaseVersion.IncPatch()
+	}
+
+	go wait()
+	fmt.Printf("VERIFICATION: Release new version '%s'? (ENTER to continue, Ctrl-C to quit))", nextReleaseVersion.String())
+	fmt.Scanln()
+
+	undoChanges := true
+	defer func() {
+		if undoChanges {
+			// git reset --hard origin/master
+			originMasterCommitHash, err := repository.ResolveRevision(plumbing.Revision(remoteMasterBranchName))
```

Comment:
1) missing `err`-checking
2) I'd recommend doing this resolve outside the `defer`, so that you're guaranteed it's good

Reply:
agreed although I realized I don't need this line though because I already retrieve this value (and error check) earlier when checking if local master and remote master are in sync.